### PR TITLE
GitHub api update.

### PR DESCRIPTION
**Pull request by 0xf8f8ff**
- incorporating changes introduced in the new version of `go-github` package
- some minor codebase changes: new GitHub option struct and updated validation methods for such structs
- updated makefile target to run all project tests at once

### CODE REVIEW

File: scm/github.go [hunk 1]
```diff
@@ -670,7 +659,7 @@ func (s *GithubSCM) RemoveMember(ctx context.Context, opt *OrgMembershipOptions)
 func (s *GithubSCM) GetUserScopes(ctx context.Context) *Authorization {
 	// Authorizations.List method will always return nil, response struct and error,
```
Comment (meling) [inline scm/github.go]: Comment not update; seems to describe only method.

File: Makefile [hunk 2]
```diff
@@ -94,7 +94,11 @@ protoset:
 
 test:
 	@cd ./web; go test
+	@cd ./web/hooks; go test
```
Comment (meling) [inline Makefile]: I think these can all be replaced with a single line:
```
go test ./...
```
Or
```
go test -v ./...
```

File: scm/github.go [hunk 3]
```diff
@@ -466,27 +465,23 @@ func (s *GithubSCM) AddTeamMember(ctx context.Context, opt *TeamMembershipOption
...
 
-	if _, _, err := s.client.Teams.AddTeamMembership(ctx, int64(opt.TeamID), opt.Username,
-		&github.TeamAddTeamMembershipOptions{Role: opt.Role}); err != nil {
-		return ErrFailedSCM{
+	if err != nil {
+		err = ErrFailedSCM{
 			GitError: err,
 			Method:   "AddTeamMember",
-			Message:  fmt.Sprintf("failed to add user %s to team ID %d with role %s", opt.Username, opt.TeamID, opt.Role),
+			Message:  fmt.Sprintf("failed to add user username: %s to team (ID %d, team name: %s) with role %s", opt.Username, opt.TeamID, opt.TeamName, opt.Role),
```
Comment (meling) [inline scm/github.go]: user username sounds weird. I like the old error msg, maybe combined with the new one. Maybe add single quote or parens around the name, e.g. `... add user (meling) to team ...`. That is, `... user (%s) to team (ID %d, ...`

File: scm/github.go (hunk 1 above)
Comment (0xf8f8ff) [reply scm/github.go]: Fixed.

File: Makefile (hunk 2 above)
Comment (0xf8f8ff) [reply Makefile]: Yeah, -v is just too vocal. Changed to `go test ./...`.

File: scm/github.go (hunk 3 above)
Comment (0xf8f8ff) [reply scm/github.go]: Fixed.

**Pull request merged**